### DISCUSSION TITLE
Renovate suggests older version for actions?

### DISCUSSION THREAD

**Discussion by kroese** [renovatebot/renovate#21891] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a Github Action, which version tag is `v4`.

On all repositories using this action, Renovate creates a pull request to update `v4` to an OLDER tag, so actually downgrading the action.

See for example: https://github.com/qemu-tools/qemu-docker/pull/70

I am trying to understand why this happens. The `v4` tag is clearly newer by several months. I even tried deleting the tag that Renovate suggested, but then it just suggest an even older one.

The tags are visible here:

https://github.com/kroese/github-release/tags

And clearly `v4` is newer than the one Renovate suggests.

So is this a bug in Renovate, does it get confused because `v4` tags get shifted by a force push each time? Or what could be the problem here?

### What you tried so far.

- Deleted all releases, so only v4 was left. Made no difference because it seems to look only at tags, not releases.
- Deleted the tags that Renovate suggested. Made no difference because it will then suggest an even older tag.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
It's using semver by default and the date ver is sorting higher. You can use allowedVersions in a package rule as a workaround for now

**Comment by kroese** [comment]
That explains a lot! I thought it would look at the age of the tag, not its name. 

Wouldn't it be a simple fix that before Renovate suggests an update for an action, it compares if the suggested version actually is pushed later than the current version?

That way you can still keep that semver logic the same, but prevent this problem.

**Reply by rarkins** [reply]
It's possible, but could cause other undesirable side effects/edge cases

**Reply by kroese** [reply]
@rarkins Then at least it should check if the tag is an officially released version on Github Marketplace. Because the v4 release was, but those old testing tags were not.

**Reply by viceice** [reply]
renovate doesn't check the market place, it only checks GitHub tags / releases. i also don't think the market place has an API to check and even it has, those would probably outdated most of the time because there is no way to automatically update it.

**Reply by kroese** [reply]
@viceice Okay.. Then the only solution would be to check if the date for the suggested tag is newer than the date for the current tag. According to @rarkins there might be some edgecases where this would not be desired. But I think the amount of cases where the current behaviour is not desired will be much larger.

**Reply by viceice** [reply]
currently not possible and pretty hard to implement
